saturn_client/core.py

```diff
@@ -0,0 +1,186 @@
+        return self._options
+
+    @property
+    def describe_sizes(self) -> Dict[str, str]:
+        """Instance size options"""
+        return self.options["sizes"]
+
+    def create_project(
+        self,
+        name: str = None,
+        description: Optional[str] = None,
+        image_uri: Optional[str] = None,
+        start_script: Optional[str] = None,
+        environment_variables: Optional[Dict] = None,
+        working_dir: Optional[str] = None,
+        workspace_settings: Optional[Dict] = None,
+    ) -> Dict[str, Any]:
+        """
+        Create a project from scratch
+
+        :param name: Name of project. This is the only field that is required.
+        :param description: Short description of the project (less than 250 characters).
+        :param image_uri: Location of the image. Example:
+            485185227295.dkr.ecr.us-east-1.amazonaws.com/saturn-dask:2020.12.01.21.10
+        :param start_script: Script that runs on start up. Examples: "pip install dask"
+        :param environment_variables: Env vars expressed as a dict. The names will be
+            coerced to uppercase.
+        :param working_dir: Location to use as working directory. Example: /home/jovyan/project
+        :param workspace_settings: Setting for the jupyter associated with the project.
+            These include: ``size``, ``disk_space``, ``auto_shutoff``, and ``start_ssh``.
+            The options for these are available from ``conn.options``.
+        """
+
+        if environment_variables:
+            environment_variables = "\n".join(
+                f"{k.upper()}={v}" for k, v in environment_variables.items()
+            )
+        try:
+            ws = WorkspaceSettings(**workspace_settings if workspace_settings else {})
+        except TypeError as err:
+            raise KeyError(
+                "The only valid workspace_setting options are: "
+                "size, disk_space, auto_shutoff, and start_ssh"
+            ) from err
+        ws.validate(self.options)
+        workspace_kwargs = ws.project_settings
+
+        project_config = {
+            "name": name,
+            "description": description,
+            "image": image_uri,
+            "start_script": start_script,
+            "environment_variables": environment_variables,
+            "working_dir": working_dir,
+            **workspace_kwargs,
+        }
+        # only send kwargs that are explicitly set by user
+        project_config = {k: v for k, v in project_config.items() if v is not None}
+
+        url = urljoin(self.url, "api/projects")
+        response = requests.post(
+            url,
+            data=json.dumps(project_config),
+            headers=self.settings.headers,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as err:
+            raise HTTPError(response.status_code, response.json()["message"]) from err
+        return response.json()
+
+
+@dataclass
+class WorkspaceSettings:
```

Comment:
@jameslamb see what you think of this guy. Eventually the backend should respond with nice errors and we can get rid of a lot of the stuff in validation.

Reply:
love this! I think it looks good.